### PR TITLE
Refactor tests

### PR DESCRIPTION
**Pull request by Orland0x**
General Housekeeping PR.

- Refactored and standardised hardhat tests 
- Renamed files to follow OZ cairo conventions
- Added contracts blueprint so increase ease of understanding the codebase 
- Updated to latest Starknet Hardhat and Devnet versions 
- Added Vanilla execution strategy for testing pruposes
- Sped up tests by using lite mode setting on devnet in the GA

### CODE REVIEW

File: contracts/ethereum/ZodiacModule/SnapshotXL1Executor.sol [hunk 1]
```diff
@@ -192,16 +192,16 @@ contract SnapshotXL1Executor is Module, SnapshotXProposalRelayer {
   ) external {
     //External call will fail if finalized proposal message was not received on L1.
     _receiveFinalizedProposal(callerAddress, proposalOutcome, executionHashLow, executionHashHigh);
-    require(whitelistedSpaces[callerAddress] == true, 'Invalid caller');
-    require(proposalOutcome != 0, 'Proposal did not pass');
-    require(_txHashes.length > 0, 'proposal must contain transactions');
+    // require(whitelistedSpaces[callerAddress] == true, 'Invalid caller');
```
Comment (Orland0x) [inline contracts/ethereum/ZodiacModule/SnapshotXL1Executor.sol]: im getting an issue with the zodiac test when it attempts to consume the message from L2, not too sure what it is. hence commented lines

Comment (Orland0x) [reply contracts/ethereum/ZodiacModule/SnapshotXL1Executor.sol]: fixed

File: README.md [hunk 2]
```diff
@@ -2,10 +2,60 @@
 
 # Snapshot X
 
-Programmable governance on StarkNet.
-
-#### [Documentation](https://docs.snapshotx.xyz)
+Programmable governance on StarkNet. Refer to the [documentation](https://docs.snapshotx.xyz) for more information.
+
+## Contracts Blueprint
+```ml
```
Comment (pscott) [inline README.md:8]: curious: what does `ml` refer to ? :p

File: README.md [hunk 3]
```diff
@@ -2,10 +2,60 @@
...
+contracts
+├─ starknet
+│   ├─ Authenticators
+│   │  ├─ EthTx.cairo — "Authenticate user via an Ethereum transaction"
+│   │  ├─ StarkTx.cairo — "Authenticate user via a StarkNet  transaction"
+│   │  ├─ StarkSig.cairo — "Authenticate user via a Starknet signature"
+│   │  └─ Vanilla.cairo — "Dummy authentication"
+│   ├─ VotingStrategies
+│   │  ├─ SingleSlotProof.cairo — "Enables the use the contents of an Ethereum storage slot as voting power"
```
Comment (pscott) [inline README.md]: `the use the contents`? ^^'

File: contracts/ethereum/ZodiacModule/SnapshotXL1Executor.sol [hunk 4]
```diff
@@ -190,11 +190,12 @@ contract SnapshotXL1Executor is Module, SnapshotXProposalRelayer {
     uint256 executionHashHigh,
     bytes32[] memory _txHashes
   ) external {
+    require(proposalOutcome != 0, 'Proposal did not pass');
+    require(_txHashes.length > 0, 'proposal must contain transactions');
+
     //External call will fail if finalized proposal message was not received on L1.
     _receiveFinalizedProposal(callerAddress, proposalOutcome, executionHashLow, executionHashHigh);
     require(whitelistedSpaces[callerAddress] == true, 'Invalid caller');
```
Comment (pscott) [inline contracts/ethereum/ZodiacModule/SnapshotXL1Executor.sol]: Why not move the this `require(whitelistedSpace...)` up above with the other ones?

**Review by pscott — APPROVED** [review]
let's gooooooo

File: README.md (hunk 2 above)
Comment (Orland0x) [reply README.md:8]: not sure

**Pull request merged**